File: backend/services/auth-service/src/main/java/com/linksphere/auth_service/dto/request/ResetPasswordRequest.java
package com.linksphere.auth_service.dto.request;

import com.linksphere.common.validation.annotation.StrongPassword;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import lombok.Data;

@Data
public class ResetPasswordRequest {

    @NotBlank(message = "Email is required")
    @Email(message = "Invalid email format")
    private String email;

    @NotBlank(message = "Token is required")
    private String otp;

    @StrongPassword(message = "Password must be strong")
    private String password;
}

File: mobile/lib/features/auth/model/reset_password_request.dart
class ResetPasswordRequest {
  final String email;
  final String otp;
  final String password;

  ResetPasswordRequest({
    required this.email,
    required this.otp,
    required this.password,
  });

  Map<String, dynamic> toJson() => {
        'email': email,
        'otp': otp,
        'password': password,
      };
}
